xml_converter/generators/generate_cpp.py

```diff
@@ -260,71 +269,73 @@ def generate_cpp_variable_data(
         cpp_includes.cpp_absolute_includes.add("type_traits")
 
     for filepath, document in sorted(data.items()):
-        fieldval = document.metadata
-        attribute_name: str = attribute_name_from_markdown_data(fieldval['name'])
+        fieldval = document.metadata_dataclass
+        attribute_name: str = attribute_name_from_markdown_data(fieldval.name)
 
-        if doc_type in fieldval['applies_to']:
+        if doc_type in fieldval.applies_to_as_str():
             xml_fields: List[str] = []
             side_effects: List[str] = []
             write_to_xml: bool = True
             default_xml_field: str = ""
 
-            if fieldval['type'] in documentation_type_data:
-                cpp_type = documentation_type_data[fieldval['type']]["cpp_type"]
-                class_name = documentation_type_data[fieldval['type']]["class_name"]
+            if fieldval.variable_type in documentation_type_data:
+                cpp_type = documentation_type_data[fieldval.variable_type]["cpp_type"]
+                class_name = documentation_type_data[fieldval.variable_type]["class_name"]
                 cpp_includes.cpp_relative_includes.add("attribute/{}.hpp".format(class_name))
 
-            elif fieldval['type'] in ["Custom", "CompoundCustomClass"]:
-                cpp_type = fieldval['class']
-                class_name = insert_delimiter(fieldval['class'], delimiter="_")
+            # elif fieldval.variable_type in ("Custom", "CompoundCustomClass"):
+            elif fieldval.variable_type == "Custom" or fieldval.variable_type == "CompoundCustomClass":
+                cpp_type = fieldval.cpp_class
+                class_name = insert_delimiter(fieldval.cpp_class, delimiter="_")
                 cpp_includes.hpp_relative_includes.add("attribute/{}.hpp".format(class_name))
 
-            elif fieldval['type'] in ["Enum", "MultiflagValue", "CompoundValue"]:
+            elif fieldval.variable_type in ("Enum", "MultiflagValue", "CompoundValue"):
```

Comment:
Is there a reason to use a Tuple here over a list?

Reply:
Conceptually no. However tuples are immutable while lists are mutable. So in theory python could do some sort of optimization. No reason to force tuples over lists but they should probably be preferred.